### ISSUE TITLE
could I got the output or processed file or the 2d to 3d, bvh file. and where?

### ISSUE THREAD

**Issue by Ai-is-light** [FORTH-ModelBasedTracker/MocapNET#3]
I added a flag : --novisualization that suppresses visualization windows

You should git pull , recompile and then try ./WebcamBIN --from ./video_test/sense_test.mp4 --novisualization and see if it will now work on the docker container..

https://github.com/FORTH-ModelBasedTracker/MocapNET/commit/d5c7471cc2e083fc1a3e21f3f7cb07c85d410836

_Originally posted by @AmmarkoV in https://github.com/FORTH-ModelBasedTracker/MocapNET/issues/2#issuecomment-526654117_

now, I can get the image imshow at docker environment.
I run root@7953ba7131a3:/pose_dancing/we-try-git-newest/MocapNET# ./WebcamBIN --from video_test/sense_test.mp4  --novisualization. Although, the code and enviroment will work, where can I got the final ouput or the 3d, 2d, bvh file?
I got as following:
Trying to open video_test/sense_test.mp4
OpenCV Error: Assertion failed (size.width>0 && size.height>0) in imshow, file /pose_dancing/MocapNET/opencv-3.2.0/modules/highgui/src/window.cpp, line 304
terminate called after throwing an instance of 'cv::Exception'
  what():  /pose_dancing/MocapNET/opencv-3.2.0/modules/highgui/src/window.cpp:304: error: (-215) size.width>0 && size.height>0 in function imshow.
Aborted

It can work for about 10 seconds or work until the end of video with the image of video_test/sense_test.mp4. However, it will end of one error nearly the ending frames or second, called, 
what():  /pose_dancing/MocapNET/opencv-3.2.0/modules/highgui/src/window.cpp:304: error: (-215) size.width>0 && size.height>0 in function imshow. Aborted. 
In fact, I build one opencv by using your  getOpenCV3.2.sh .
 I'm a little bit confuesed with it. If l don't build the opnecv by getOpenCV3.2.sh, just using the pip install opencv-python, what will happend? Anyway, with the old version of this repo, I got the result out keypoints and others, but the old version can only work with the first some frames, I will be Aborted, which may be caused by so many windows with in dokcer.

And, there the latest results. Thanks for your updating and replies.
![Uploading image.png…]()

> I added a flag : --novisualization that suppresses visualization windows
> 
> You should git pull , recompile and then try ./WebcamBIN --from ./video_test/sense_test.mp4 --novisualization and see if it will now work on the docker container..
> 
> [d5c7471](https://github.com/FORTH-ModelBasedTracker/MocapNET/commit/d5c7471cc2e083fc1a3e21f3f7cb07c85d410836)

now, I can get the image imshow at docker environment.
and, --novisualizaition man not work?
I run root@7953ba7131a3:/pose_dancing/we-try-git-newest/MocapNET# ./WebcamBIN --from video_test/sense_test.mp4  --novisualization
I got as following:
Trying to open video_test/sense_test.mp4
OpenCV Error: Assertion failed (size.width>0 && size.height>0) in imshow, file /pose_dancing/MocapNET/opencv-3.2.0/modules/highgui/src/window.cpp, line 304
terminate called after throwing an instance of 'cv::Exception'
  what():  /pose_dancing/MocapNET/opencv-3.2.0/modules/highgui/src/window.cpp:304: error: (-215) size.width>0 && size.height>0 in function imshow.
Aborted

It can work for about 10 seconds with the image of video_test/sense_test.mp4. However, nothing have been processed. I'm a little bit confuesed with it. Anyway, with the old version of this repo, I got the result out keypoints and others, but the old version can only work with the first some frames, I will be Aborted, which may be caused by so many windows with in dokcer.

**Comment by AmmarkoV** [comment]
The BVH file default filename is out.bvh ( except if you override it with -o filename.bvh ) 

The BVH file is written after the code is done with the main loop using the writeBVHFile call  https://github.com/FORTH-ModelBasedTracker/MocapNET/blob/master/WebcamAndDeepJoint/test.cpp#L520 This is why if you get an error there is no file generated.

I have added logic in the OpenCV code so that it anticipates empty input frames that might be present in a file due to encoding problems.

Please  git pull , recompile and then try again and tell me if the issue is fixed..!

**Comment by Ai-is-light** [comment]
after updating and rebuild, I also got some odd error as following:
it only can show and process several sconds

Tensor: result_back/concat inputs: 133
 outputs: 1
Caching networks after initialization to be ready for use..
2019-09-04 13:56:33.357048: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.
LoadGraph combinedModel/vnect_sm_pafs_8.1k.pb using TensorFlow Version: 1.14.0
 Input Tensor for combinedModel/vnect_sm_pafs_8.1k.pb
Tensor: input_1 inputs: 0
 outputs: 1
 Output Tensor for combinedModel/vnect_sm_pafs_8.1k.pb
Tensor: k2tfout_1 inputs: 1
 outputs: 1
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 0
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 1
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 2
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 3
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 4
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 5
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 6

(BGR:24176): GLib-GObject-CRITICAL **: g_object_unref: assertion 'G_IS_OBJECT (object)' failed
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 7
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 8
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 9
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 10
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 11
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 12
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 13
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 14
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
MocapNET: Incorrect size of COCO input  was 168 (but should be 171)
Won't visualize empty neural network output for frame 15
Haven't detected a person, so seeking a skeleton in the full image, regardless of distortion..
Direction is : 13.9689 Front
Right Shoulder Z X Y = 64.46,24.13,9.98
Left Shoulder Z X Y = -72.50,21.80,-11.57
buildOpenGLProjectionForIntrinsics: Size(1024x768) fx 570.00, fy 570.00, cx 512.00, cy 384.00, skew 0.00, Near 1.00, Far 10000.00
Could not calculate torso
Area does not exist, why get 2D positions?
buildOpenGLProjectionForIntrinsics: Size(1024x768) fx 570.00, fy 570.00, cx 512.00, cy 384.00, skew 0.00, Near 1.00, Far 10000.00
OpenCV Error: Assertion failed (ssize.width > 0 && ssize.height > 0) in resize, file /pose_dancing/MocapNET/opencv-3.2.0/modules/imgproc/src/imgwarp.cpp, line 3492
terminate called after throwing an instance of 'cv::Exception'
  what():  /pose_dancing/MocapNET/opencv-3.2.0/modules/imgproc/src/imgwarp.cpp:3492: error: (-215) ssize.width > 0 && ssize.height > 0 in function resize

Aborted
root@7953ba7131a3:/pose_dancing/we-try-git-newest/MocapNET-0904#

**Comment by AmmarkoV** [comment]
Unfortunately the error still appears to be OpenCV related and has to do with your input source..!

As seen from :
what(): /pose_dancing/MocapNET/opencv-3.2.0/modules/imgproc/src/imgwarp.cpp:3492: error: (-215) ssize.width > 0 && ssize.height > 0 in function resize

which is probably emitted @ https://github.com/FORTH-ModelBasedTracker/MocapNET/blob/master/WebcamAndDeepJoint/test.cpp#L513 while trying to resize the input image to fit the 368x368 input size of the 2D detector.

The best idea I can offer is to try 
wget http://ammar.gr/mocapnet/shuffle.webm 
This will download a sample video that is encoded using the opensource webm format ( so to try to overcome a problem that has to do with missing codecs? )
 
Then trying to ./WebcamJointBIN --from shuffle.webm --novisualization --frames 377
This should work and should output a file called out.bvh with the results..

**Comment by AmmarkoV** [comment]
Closing this issue as it appears to be stale
